core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterBuilderService.java

```diff
@@ -134,15 +134,13 @@ public void buildCluster(Long stackId) throws CloudbreakException {
         clusterService.updateCluster(cluster);
         final Telemetry telemetry = componentConfigProviderService.getTelemetry(stackId);
 
-        if (cluster.getProxyConfigCrn() != null) {
-            ProxyConfig proxyConfig = proxyConfigDtoService.getByCrn(cluster.getProxyConfigCrn());
-            if (proxyConfig != null) {
-                LOGGER.info("proxyConfig is not null, setup proxy for cluster");
-                connector.clusterSetupService().setupProxy(proxyConfig);
-            } else {
-                LOGGER.info("proxyConfig was not found by proxyConfigCrn");
-            }
-        }
+        Optional<ProxyConfig> proxyConfig = proxyConfigDtoService.getByCrnWithEnvironmentFallback(cluster.getProxyConfigCrn(), cluster.getEnvironmentCrn());
+        proxyConfig.ifPresentOrElse(
+                pc -> {
+                    LOGGER.info("proxyConfig is not null, setup proxy for cluster");
+                    connector.clusterSetupService().setupProxy(pc);
+                },
```

Comment:
move this into a method please

Reply:
I removed all the proxy setup part to a different method. Is it ok?